fe/fe-core/src/main/java/org/apache/doris/transaction/DatabaseTransactionMgr.java

```diff
@@ -2290,4 +2480,177 @@ protected void updateMultiTableRunningTransactionTableIds(long transactionId, Li
+                //  2. last failed version < 0: is a health replica before
+                //  3. version catch up: not with a stale version
+                // Here we only check number, the replica version will be updated in updateCatalogAfterVisible()
+                for (MaterializedIndex index : allIndices) {
+                    for (Tablet tablet : partition.getIndex(index.getId()).getTablets()) {
+                        List<Replica> tabletSuccReplicas = Lists.newArrayList();
+                        List<Replica> tabletWriteFailedReplicas = Lists.newArrayList();
+                        List<Replica> tabletVersionFailedReplicas = Lists.newArrayList();
+                        // TODO always use the visible version because the replica version is not changed
+                        long newVersion = partition.getVisibleVersion() + 1;
+                        for (Replica replica : tablet.getReplicas()) {
+                            for (PublishVersionTask publishVersionTask : publishTasks.get(replica.getBackendId())) {
+                                boolean needCheck = publishVersionTask.getTransactionId()
+                                        == subTransactionState.getSubTransactionId()
+                                        && publishVersionTask.getPartitionVersionInfos().stream()
+                                        .anyMatch(s -> s.getPartitionId() == partitionId);
+                                if (needCheck) {
+                                    checkReplicaContinuousVersionSucc(tablet.getId(), replica, alterReplicaLoadedTxn,
+                                            newVersion, publishVersionTask,
+                                            errorReplicaIds, tabletSuccReplicas, tabletWriteFailedReplicas,
+                                            tabletVersionFailedReplicas);
+                                    LOG.debug("after checkReplicaContinuousVersion for txn_id={}, sub_txn_id={}, "
+                                                    + "tablet_id={}, new_version={}, success_replicas={}, "
+                                                    + "write_failed_replicas={}, version_failed_replicas={}",
+                                            transactionState.getTransactionId(),
+                                            subTransactionState.getSubTransactionId(), tablet.getId(), newVersion,
+                                            tabletSuccReplicas, tabletWriteFailedReplicas, tabletVersionFailedReplicas);
+                                }
+                            }
+                        }
+
+                        publishResult = checkQuorumReplicas(transactionState, tableId, partition, tablet,
+                                loadRequiredReplicaNum, allowPublishOneSucc, newVersion, tabletSuccReplicas,
+                                tabletWriteFailedReplicas, tabletVersionFailedReplicas, publishResult, logs);
+                    }
+                }
+            }
+        }
+
```

Comment:
Here, we need assure that there is no hole at least for one replica.

Reply:
support it and add a ut `GlobalTransactionMgrTest#testCommitTransactionWithSubTxnAndReplicaFailed` check it.